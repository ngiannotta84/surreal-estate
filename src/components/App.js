import React from "react";
import { Routes, Route } from "react-router-dom";
import "../styles/app.css";
import AddProperty from "./AddProperty";
import Navbar from "./Navbar";
import Properties from "./Properties";

const App = () => {
  return (
    <div className="App">
      <h2>Surreal Estate</h2>
      <Navbar />
      <Routes>
        <Route path="/" element={<Properties />} />
        <Route path="add-property" element={<AddProperty />} />
      </Routes>
    </div>
  );
};
/* it is Navabar that is displaying the links the Routes are just defing the paths that they need 
to follow and then diaplyin the actual components */

export default App;
