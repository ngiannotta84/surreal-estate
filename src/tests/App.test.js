import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

test("renders Surreal Estate", () => {
  render(<App />);
  const linkElement = screen.getByText(/surreal estate/i);
  expect(linkElement).toBeInTheDocument();
});
