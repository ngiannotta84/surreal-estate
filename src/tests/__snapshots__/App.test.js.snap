// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h2>
      Surreal Estate
    </h2>
    <div
      class="navbar"
    >
      <img
        alt="logo of Surreal Estate"
        class="logo"
        data-testid="logo"
        src="logo.png"
      />
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-links-item"
        >
          <a
            class="item"
            href="/"
          >
            View Property
          </a>
        </li>
        <li
          class="navbar-links-item"
        >
          <a
            class="item"
            href="/add-property"
          >
            Add a Property
          </a>
        </li>
      </ul>
    </div>
    <div>
      Properties page
    </div>
  </div>
</DocumentFragment>
`;
